File: iniciante/python/1018-cedulas.py
def entrada_valida():
    numero = int(input())
    if numero <= 0 or numero >= 1000000:
        while True:
            numero = int(input())
            if 0 < numero < 1000000:
                break
    return numero


def cedulas(n):
    print(n)
    notas = [100, 50, 20, 10, 5, 2, 1]
    restante = n
    for nota in notas:
        cedula = restante // nota
        restante -= cedula * nota
        print(f'{cedula} nota(s) de R$ {nota},00')


cel = entrada_valida()
cedulas(cel)

File: iniciante/c/1018-cedulas.c
#include <stdio.h>

int main(){
    int numero, i, resto, quantidade;
    int notas[6] = {100, 50, 20, 10, 5, 2};   
    
    scanf("%d", &numero);

    resto = numero;
    for(i = 0; i < 6; i++){
        quantidade = resto / notas[i];
        
        printf("%d nota(s) de R$ %d,00\n", quantidade, notas[i]);

        resto = resto % notas[i];
    }

    printf("%d nota(s) de R$ 1,00\n", resto);
    
    return 0;
}
